src/core/handlers/RequestHandler.ts

```diff
@@ -30,7 +30,7 @@ export interface RequestHandlerInternalInfo {
 export type ResponseResolverReturnType<
   BodyType extends DefaultBodyType = undefined,
 > =
-  | (BodyType extends undefined ? Response : StrictResponse<BodyType>)
```

Comment:
`T extends undefined` is not the correct way to check if a generic was provided in TypeScript. Instead, one must resort to a hack:

```ts
[T] extends [never] ? 'NOT PROVIDED' : 'PROVIDED'
```

The previous implementation was giving false positives, making the custom response body type and the body type in `ResponseResolverReturnType` incompatible.